internal/databases/clickhouse/backup_push_handler.go

```diff
@@ -0,0 +1,91 @@
+package clickhouse
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/utility"
+	"github.com/wal-g/wal-g/internal"
+)
+
+func HandleBackupPush(uploader *internal.Uploader, backupCmd *exec.Cmd) {
+	backupFolderRootPath, _ := internal.GetSetting(internal.ClickHouseBackupPath)
+
+	backupName := getBackupName()
+	appendBackupNameToCommand(backupCmd, backupName)
+	_, stderr, err := utility.StartCommandWithStdoutStderr(backupCmd)
+	if err := backupCmd.Wait(); err != nil {
+		tracelog.ErrorLogger.Printf("backup command output:\n%s", stderr.String())
+		tracelog.ErrorLogger.Fatalf("failed to run backup: %v", err)
+	}
+
+	backupFolderPath := path.Join(backupFolderRootPath, backupName)
+	filePaths, err := getBackupFilePaths(backupFolderPath)
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("failed to get backup filepaths: %v", err)
+	}
+
+	for _, filePath := range filePaths {
+		err = uploadBackupFile(uploader, backupFolderRootPath, filePath)
+		if err != nil {
+			tracelog.ErrorLogger.Fatalf("failed to push file: %v", err)
+		}
+	}
```

Comment:
After the finish of the uploading process, WAL-G needs to upload a sentinel file with some basic metadata about the created backup. You can see how it is [implemented in MySQL](https://github.com/wal-g/wal-g/blob/884a6c503ce92ffe6377393b01272580f06117df/internal/databases/mysql/backup_push_handler.go#L53) and use it as a reference.